libp2p/stream_muxer/mplex/mplex_stream.py

```diff
@@ -34,13 +37,33 @@ def __init__(self, stream_id: int, initiator: bool, mplex_conn: IMuxedConn) -> N
         self.local_closed = False
         self.remote_closed = False
         self.stream_lock = asyncio.Lock()
+        self._buf = None
 
-    async def read(self) -> bytes:
+    async def read(self, n: int = -1) -> bytes:
         """
         read messages associated with stream from buffer til end of file
+        :param n: number of bytes to read
         :return: bytes of input
         """
-        return await self.mplex_conn.read_buffer(self.stream_id)
+        if n == -1:
```

Comment:
Should we check `if self._buf is None` first before reading from `mplex_conn`?

Reply:
Yes, you're right. Good catch. I will actually replace `read` with `read_bytes` later.